browser/src/control/Control.NavigatorPanel.ts

```diff
@@ -421,6 +449,11 @@ class NavigatorPanel extends SidebarBase {
 				builder,
 			);
 		}
+
+		const quickFindTab = this.navigationPanel.querySelector('#tab-quick-find');
+		if (quickFindTab.classList.contains('selected')) {
```

Comment:
We should avoid qyerying the DOM if possible.
Did you try to just put it inside above block `} else if (object.id === 'navigator-search') {` ?
`navigator-search` is a search field, you can detect on any event of it if field is empty.
I belive you can just check if data/searchTerm is empty or not there.